### ISSUE TITLE
[BUG] door.is_subhint behavior change for `typing` generic types

### ISSUE THREAD

**Issue by mthramann** [beartype/beartype#461]
Thanks for the excellent library!! I noticed a change in behavior from 0.18.5 to 0.19.0 and it looks likes a regression to me, but perhaps there's a subtlety I've missed.

On 0.19.0:
```py
from beartype.door import is_subhint
from typing import TypeVar, Generic, Sequence

T = TypeVar("T")

class A(Generic[T]): ...

class B(A[T]): ...

assert is_subhint(B[int], A[int])  # Okay

class MySequence(Sequence[T]): ...

assert is_subhint(MySequence[int], Sequence[int])  # Boom
```

This passes fine on 0.18.5. Is this an intended change in behavior? Admittedly I have not checked the other generic types in `typing`. Thank you!

**Comment by leycec** [comment]
**Gah!** That's demonstrably awful. Thanks so much for catching this. @beartype sports an extensive test suite to catch regressions like this "out in the open," but it looks like this edge case somehow slipped past us. What a ninja issue.

We'll get this patched up in a jiffy for our upcoming @beartype 0.19.1 patch release. This is why I sigh in the morning. :face_exhaling:

**Comment by Moosems** [comment]
In fairness leycec, never expect to catch everything in your test suite. There will always be users who find something unexpected

**Comment by leycec** [comment]
**...lol.** I have good news and I have bad news. The:

* Good news is that this *never* worked. This has been broken for over a year (although it feels much longer)! Rejoice, everyone. Broken things have stayed broken rather than magically working when somebody wasn't looking.
* Bad news is that this is a duplicate of #271 previously reported by gentleman QAer @patrick-kidger. Let's close out this clone in favour of the original. Poor clone. :disappointed:

**Comment by leycec** [comment]
**lol** x 2. So, I have no idea what I'm talking about. This *did* genuinely broken due to insufficient regression testing. Reopen, GitHub! Reopen Sesame. :door:

**Comment by leycec** [comment]
**Resolved by dd9efc24876939e.** This never happened. You saw nothing. The Bears in Black (BIB) were never here.

![](https://media1.tenor.com/m/ow0-gzBN9oIAAAAd/burrito-bear-spaceorca.gif)
<sup>*@beartype 0.19.0 don't know nuffin'*</sup>
